sympy/sets/tests/test_sets.py

```diff
@@ -1145,3 +1145,24 @@ def test_issue_16878b():
     # that handles the base_set of S.Reals like there is
     # for Integers
     assert imageset(x, (x, x), S.Reals).is_subset(S.Reals**2) is True
+
+
+def test_issue_17274():
+    neinf = S.NegativeInfinity
+    poinf = S.Infinity
+    set_ff = Interval(0, 5)
+    set_if = Interval(-oo, 5)
+    set_fi = Interval(0, oo)
+    set_ii = Interval(-oo, oo)
+    fs_ni = FiniteSet(S.NegativeInfinity)
+    fs_pi = FiniteSet(S.Infinity)
+
+    assert set_ff + neinf == fs_ni
+    assert set_ff + poinf == fs_pi
+    assert set_fi + neinf == set_ii
+    assert set_if + poinf == set_ii
+
+    assert set_ff - poinf == fs_ni
+    assert set_ff - neinf == fs_pi
+    assert set_if - neinf == set_ii
+    assert set_fi - poinf == set_ii
```

Comment:
These tests may fail because they are not wrapped in `SetExpr`.
You may have to wrap everything with `SetExpr`, and also put in the setexpr test file, unless you're going to test about unions here.

Reply:
The suggestion, is want I have done before, however, they all failed. I still need more research on this section.

Reply:
What do you get if you do `repr(set_ff - poinf)` or `srepr(set_ff - poinf)`? The latter is the full SymPy expression, but most of the time the `repr` one is good enough to understand the result. At least that is the result that SymPy currently returns and it is at least tested for syntax errors etc. Ideally it should also be the correct one...

Reply:
@oscargus , @sylee957 , Problem solved, it seems there is a tiny problem in the implementation in _set_sub before. @sylee957 , you are right, it should be wrapped by SetExpr.